doc/ansible/dev/developer_guide.md

```diff
@@ -390,6 +390,22 @@ can be used as shown:
       foo: bar
 ```
 
+### Ansible Operator Conditions
+The Ansible Operator has a set of conditions which it will use as it performs
+its reconciliation procedure. There are only a few main conditions:
+
+* Running - the Ansible Operator has called Run on the Ansible Runner to invoke
+  your Ansible logic.
+
+* Successful - if the run has finished and there were no errors, the Ansible
+  Operator will be marked as Successful. It will then wait for the next
+  reconciliation period.
+
+* Failed - if there is any error during the reconcilation run, the Ansible
```

Comment:
Can we make a note about the failure message and that it comes from ansible? 

I also think that it might be helpful to link to the debug section here (if we don't have one we should find all the different parts throughout our docs and put together one IMO). WDYT?

Reply:
I agree we should have a new debug section, for now I simply linked to the other areas I've found that have relevant information. I think the debug should be a separate PR.